com.unity.sg2/Editor/GraphDeltaRegistry/Generation/Interpreter.cs

```diff
@@ -363,6 +364,13 @@ private static bool FunctionsAreEqual(ShaderFunction a, ShaderFunction b)
                                 if (tex != null && !defaultTextures.Contains((name, tex)))
                                     defaultTextures.Add((name, tex));
                             }
+
+                            if (port.GetTypeField().GetRegistryKey().Name == SamplerStateType.kRegistryKey.Name)
+                            {
+                                var fieldHandler = port.GetTypeField();
+                                var field = SamplerStateType.UniformPromotion(port.GetTypeField(), container, registry);
+                                inputVariables.Add(field);
+                            }
```

Comment:
Textures (above) and Samplers are the only cases where an an inline value has to be promoted to a material property. I'll consider abstracting this if we run into situations where that will be helpful, but for now I'm not going to fiddle with this much further.